scripts/powershell/sub-cleanup/sub-cleanup.ps1

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env pwsh -c
+
+<#
+.SYNOPSIS
+Subscription cleanup script
+
+.DESCRIPTION
+Cleans up non-compliant resource groups in a specified Azure
+subscription.  Complient resource groups are named with a valid
+Microsoft alias or using the format: alias-<identifier>
+
+.NOTES
+You can add the tag CleanupException=true to the resource group
+to have it skipped as an exception. This can be done with CLI
+or in the portal.
+
+Requires PowerShell module Az and RemoteSigned excecution policy.
+Commands to make this happen:
+  PS> Install-Module -Name Az -AllowClobber
+  PS> Set-ExecutionPolicy RemoteSigned
+
+.EXAMPLE
+PS> sub-cleanup.ps1 -Subscription {guid} -Audit false
+
+.PARAMETER Subscription
+Subscription ID (guid) for the subscription to be cleaned.
+
+.PARAMETER Audit
+When specified, deletion operations are just logged, but not performed.
+#>
+
+param(
+  [Parameter(Mandatory = $true)]
+  [string]$Subscription,
+  
+  [Parameter(Mandatory = $false)]
+  [boolean]$Audit=$true
+)
+
+# Check if we're in audit mode (no actual deletions)
+if ($Audit)
+{
+    Write-Host -ForegroundColor Green "Running in audit mode. Nothing will be deleted."
+}
+else
+{
+    Write-Host -ForegroundColor Red "!!!NOT IN AUDIT MODE. RESOURCE GROUPS WILL BE DELETED!!!"
+}
+
+# Connect and select active subscription
+Connect-AzAccount
```

Comment:
https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/TestResources/Remove-TestResources.ps1#L8

Might want to add some requires clauses to ensure you have the Az modules you are using.

Reply:
On top of that, unless you go with what Mitch was proposing, I recommend copying the brunt of the auth code from the TestResources scripts linked above. May not need it all since you're likely not supporting non-automation for this script, but you'd still want to log in as a provisioner. Or is this for interactive?